(L)   Update to transfer transaction building, and signing.

mseaward opened the pull request:
Manipulated the building of the transfer transaction as keys required are not available in their old location when using ScatterDesktop in conjunction with ScatterBridge and ScatterWalletpack.

There is now separate function for building the TransactionBuilder transaction memo object. Temporary object properties are used such that required data can be passed along to other functions without breaking the scope of existing "core" functions. ie: the memo recipient, the memo itself, etc.

mseaward committed:
Refactor to more maintainable code.
- Scatter UI specific code left in the root plugin and helper sub-lib is within another file and exports helper functions.

mseaward committed:
Hardcode assetId into getTransferTransaction as the Scatter plugin is only to support peerplays' core uia

mseaward committed:
Add function to convert a human readable num to a blockchain num based on token/asset precision. Added docs to new function. Modified unit tests to support changes made.

MuffinLightning committed:
Merge pull request #37 from peerplays-network/bug/ui-num-convert

Add function to convert human readable num to blockchain num

MuffinLightning committed:
CORS fix

mseaward committed:
Added methods to support storing all three authority level Peerplays WIF keys under a single Scatter KeyPair instance.
- a new "master key" will be generated from the encrypted result of all three WIFs, in object format, and be set as the Scatter KeyPair.privateKey
1. form gets username & password
2. peerplaysjslib.Login.generateKeys(username, password)
3. WIFs = convert keys from step 2 into WIFs
4. encrypt JSON.stringify(WIFs)
5.1. store result of 4 under Scatter KeyPair under privateKey
5.2 store WIFs.owner converted to publoc key string uner Scatter KeyPair under prublicKeys[0].key

Relates to #26, #16

mseaward committed:
Merge remote-tracking branch 'origin/peerplays-plugin' into peerplays-one-keypair

mseaward committed:
Merge pull request #38 from peerplays-network/peerplays-one-keypair

[L] Peerplays one keypair with all WIFs within it

MuffinLightning committed:
Peerplays transfer updates for ui support

MuffinLightning committed:
readded chain id

MuffinLightning committed:
Merge pull request #42 from peerplays-network/peerplays-transfer-updates

Peerplays transfer updates for ui support

mseaward committed:
enhanced documentation on some functions within PPYKeypairService.js

MuffinLightning committed:
Update for importing, storing, decrypt

MuffinLightning committed:
Merge branch 'peerplays-plugin' into peerplays-import

MuffinLightning committed:
indenting

MuffinLightning committed:
requested changes

MuffinLightning committed:
Fixed doc.

MuffinLightning committed:
Changed logs to err

MuffinLightning committed:
Documentation :^)

MuffinLightning committed:
Added unit test for import

mseaward committed:
Merge pull request #45 from peerplays-network/peerplays-import

Peerplays Import

MuffinLightning committed:
Memo support

mseaward committed:
Remove dead/unused code

MuffinLightning committed:
Signing logic update

mseaward committed:
revert back to KeyPairService.publicToPrivate instead of the duplicate function added.

mseaward committed:
Remove cryptography on PPY "master" key and use simple encode/decode with buffers. Refactor variable name in Peerplays.js

mseaward committed:
krse decode in PPYKeypairService

mseaward committed:
Handle deprecation error for `new Buffer`

mseaward committed:
Manipulated the building of the transfer transaction as keys required are not available in their old location when using ScatterDesktop in conjunction with ScatterBridge and ScatterWalletpack.

There is now separate function for building the TransactionBuilder transaction memo object. Temporary object properties are used such that required data can be passed along to other functions without breaking the scope of existing "core" functions. ie: the memo recipient, the memo itself, etc.

mseaward committed:
Merge branch 'peerplays-plugin' of github.com:peerplays-network/getscatter-walletpack into peerplays-new-transfer

mseaward committed:
remove unused var

mseaward committed:
provide object metadata within doc block for transfer

MuffinLightning reviewed (changes requested):
bufferToHexPrivate needs to be updated.

pubActiveKey is not properly set within signer()

mseaward committed:
Modufy bufferToHexPrivate to work with our data

mseaward committed:
Fix unit tests: #50
- alter peerplays.test.js tests to use the dummy environment setup that is used within the [walletpack sample test suite](https://github.com/GetScatter/walletpack/blob/fio/packages/fio/__tests__/fio.test.js)
- write new transfer tests
- remove tests that would be tested as a bydproduct of the transfer test (if transfer test passes, they would too)

mseaward committed:
Merge tests to reduce duplicate code (sonarcloud errors)

mseaward committed:
add doc block to pkrivateToPublic
p ys.Please enter the commit message for your changes. Lines starting
